pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensions/channel/ServiceUnitStateChannelImpl.java

```diff
@@ -487,6 +487,27 @@ public boolean isOwner(String serviceUnit) {
         return isOwner(serviceUnit, lookupServiceAddress);
     }
 
+    private CompletableFuture<Optional<String>> getActiveOwnerAsync(
```

Comment:
Should we return the `Optional<BrokerLookupData>` here so we don't need to use `getOwnershipWithLookupDataAsync` in the `ExtensibleLoadManagerImpl`

Reply:
I think we can keep `getOwnershipWithLookupDataAsync` as-is, as we check the broker registry at the above ExtensibleLoadManagerImpl layer. This is a post-validation right before the return without deferring. I think this additional check is harmless, as it is mostly memory lookup.

Reply:
I'm worried that before `getOwnershipWithLookupDataAsync` broker are shutdown, it will still get Failed to look up a broker

Reply:
Upon broker shutdown, 
here we think of two lookup cases.

Case 1:

getOwnerAsync should defer the lookup if that broker is not found in the registry. 


Case 2:

getOwnerAync returns the shutdown broker if that broker is still found in the registry.

In this case, we have no choice. Clients should retry the lookup.

Reply:
Yes, the clients should retry.